Source/hydro/Castro_ctu_hydro.cpp

```diff
@@ -1311,110 +1307,95 @@ Castro::construct_ctu_hydro_source(Real time, Real dt)
 #endif
 
         if (idir == 0) {
+#if AMREX_SPACEDIM <= 2
+            Array4<Real> pradial_fab = pradial.array();
+#endif
+
             // get the scaled radial pressure -- we need to treat this specially
+#if AMREX_SPACEDIM <= 2
+
 #if AMREX_SPACEDIM == 1
             if (!Geom().IsCartesian()) {
-                amrex::ParallelFor(nbx,
-                [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k) noexcept
-                {
-                    pradial_fab(i,j,k) = qex_arr(i,j,k,GDPRES) * dt;
-                });
-            }
-#endif
-
-#if AMREX_SPACEDIM == 2
+#elif AMREX_SPACEDIM == 2
             if (!mom_flux_has_p(0, 0, coord)) {
+#endif
                 amrex::ParallelFor(nbx,
                 [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k) noexcept
                 {
                     pradial_fab(i,j,k) = qex_arr(i,j,k,GDPRES) * dt;
                 });
             }
+
 #endif
         }
 
-        // Store the fluxes from this advance.
+        // Store the fluxes from this advance. For simplified SDC integration we
+        // only need to do this on the last iteration.
 
-        // For normal integration we want to add the fluxes from this advance
-        // since we may be subcycling the timestep. But for simplified SDC integration
-        // we want to copy the fluxes since we expect that there will not be
-        // subcycling and we only want the last iteration's fluxes.
+        bool add_fluxes = true;
 
-        Array4<Real> const flux_fab = (flux[idir]).array();
-        Array4<Real> fluxes_fab = (*fluxes[idir]).array(mfi);
-        const int numcomp = NUM_STATE;
+        if (time_integration_method == SimplifiedSpectralDeferredCorrections &&
+            sdc_iteration == sdc_iters - 1) {
+            add_fluxes = false;
```

Comment:
should this be `sdc_iteration != sdc_iters - 1`?

Reply:
Yes, fixed.